superset/reports/commands/update.py

```diff
@@ -87,6 +88,11 @@ def validate(self) -> None:
         # Validate chart or dashboard relations
         self.validate_chart_dashboard(exceptions, update=True)
 
+        if "validator_config_json" in self._properties:
+            self._properties["validator_config_json"] = json.dumps(
+                self._properties["validator_config_json"]
+            )
+
```

Comment:
wondering if we should handle this client side before we make the request. It is confusing having json inside json. Or perhaps add a check to see if `validator_config_json` is already a json string before running this, that way the api would support both (json string or json object).

Reply:
@dpgaspar added this logic to make it easier for the FE (you can see it in the `create` logic already), by not requiring us to stringify our json beforehand, which I'm fine with. Do you see it as being an issue in the long run?

Reply:
My main concern is with the api spec (ie, `/swagger/v1`). If it specifies a json string for this field a user will likely make an api request with a json string and possibly encounter a opaque error message from the api server.

Reply:
I think that the OpenAPI spec is clear on this, it's just a nested structure, this happens on other places also. I think it's more elegant then sending a JSON string for a specific field